### PR TITLE
store: guests will only receive results of which channels they are in

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Do not allow guest accounts discover public channels if they are not added to those channels in autocomplete.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44575

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by isacikgoz** [comment]
> This needs to be done with Bleve and elasticsearch as well. Were you planning to send PRs after this?

Ah, I totally missed that.. 🤦 Will include in to this one. Thanks!

**Review by mkraft — APPROVED** [review]
LGTM. I couldn't poke any holes.

File: services/searchengine/bleveengine/search.go [hunk 1]
```diff
@@ -329,6 +329,13 @@ func (b *BleveEngine) SearchChannels(teamId, userID, term string) ([]string, *mo
 	// AND
 	// match (channelType != 'P' || (<> in userIDs && channelType == 'P'))
 
+	// (or if isGuest is true)
```
Comment (agnivade) [inline services/searchengine/bleveengine/search.go:332]: Hopefully my comments were helpful to figure out the logic.

**Review by agnivade — APPROVED** [review]

File: services/searchengine/bleveengine/search.go (hunk 1 above)
Comment (isacikgoz) [reply services/searchengine/bleveengine/search.go:332]: Those were top class comments, definitely helped a lot :)

**Review by jupenur — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
Working as expected and E2E passed
1. Log in as Sysadmin
2. Create `PUBLIC` channel A , B and C
3. Enter person’s email address (e.g. john@example.com), select “Invite as Guest”
6. Check your email inbox and sign up to the Mattermost instance
7. Verify your account
8. Press CTRL/CMD+K to open channel switcher
9. Type C 
10. Guest should only see the channels they were added
11. image

**Comment by amyblais** [comment]
@isacikgoz This needs manual cherry-picks to 6.7, 6.6, 6.5 and 6.3 branches, and the 7.0 cherry-pick has failing tests.

**Comment by amyblais** [comment]
@isacikgoz Ping me in this PR when you're done.

**Comment by isacikgoz** [comment]
ping @amyblais

**Pull request merged**